src/signer/default-signer.ts

```diff
@@ -0,0 +1,147 @@
+import { TransactionBody } from "@ainblockchain/ain-util";
+import Provider from "../provider";
+import Wallet from "../wallet";
+import { Signer } from "./signer";
+import { TransactionInput } from "../types";
+import Ain from "../ain";
+
+/**
+ * The default class of Signer interface implemented using Wallet class.
+ * When Ain class is initialized, DefaultSigner is set as its signer.
+ */
+export class DefaultSigner implements Signer {
+  readonly wallet: Wallet;
+  readonly provider: Provider;
+
+  /**
+   * Initializes the class.
+   * @param {Wallet} wallet - The wallet to initialize with.
+   */
+  constructor(wallet: Wallet, provider: Provider) {
+    this.wallet = wallet;
+    this.provider = provider;
+  }
+
+  /**
+   * Returns the checksum address to sign messages with.
+   * If the address is not given, the default address of the wallet is used.
+   * @param {string} address - The address of the account to sign the message with.
+   */
+  getAddress(address?: string): string {
+    return this.wallet.getImpliedAddress(address);
+  }
+
+  /**
+   * Signs a message with the private key of the given address.
+   * If an address is not given, the default address of the wallet is used.
+   * @param {string} message - The message to sign.
+   * @param {string} address - The address of the account to sign the message with.
+   */
+  signMessage(message: string, address?: string): Promise<string> | string {
+    return this.wallet.sign(message, address);
+  }
+
+  /**
+   * Signs and sends a transaction to the network
+   * @param {TransactionInput} transactionObject
+   * @param {boolean} isDryrun - dryrun option.
+   * @return {Promise<any>}
+   */
+  async sendTransaction(transactionObject: TransactionInput, isDryrun: boolean = false) {
+    const txBody = await this.buildTransactionBody(transactionObject);
+    const signature = await this.wallet.signTransaction(txBody, transactionObject.address);
+    return await this.sendSignedTransaction(signature, txBody, isDryrun);
+  }
+
+  /**
+   * Sends signed transactions to the network.
+   * @param {TransactionInput[]} transactionObjects
+   */
+  async sendTransactionBatch(transactionObjects: TransactionInput[]): Promise<any> {
+    let promises: Promise<any>[] = [];
+    for (let tx of transactionObjects) {
+      promises.push(this.buildTransactionBody(tx).then(async (txBody) => {
```

Comment:
question) Is this the only place where buildTransactionBody() is used?

Reply:
In this code, `buildTransactionBody` is used only in `sendTransaction` and `sendTransactionBatch`.

Reply:
Thanks for the confirmation!